timemachine/cpp/src/context.cu

```diff
@@ -33,12 +33,17 @@ Context::Context(
 
     // A no-op if running in vacuum or there are no NonbondedAllPairs potentials
     get_nonbonded_all_pair_potentials(bps, nonbonded_pots_);
+
+    // Construct a stream unique to the Context
+    gpuErrchk(cudaStreamCreateWithFlags(&stream_, cudaStreamNonBlocking));
```

Comment:
clarification: were we previously only using stream 0 in the context implementation? (I think this is what

```cxx
cudaStream_t stream = static_cast<cudaStream_t>(0);
```

indicates, but not sure...)

Reply:
Are there any drawbacks to constructing additional streams, besides construction time? (I'm assuming the latter is why we don't e.g. construct a new stream at the beginning of every context member function implementation)

Reply:
Correct, we were using the default stream (no strictly stream 0) and when parallelized all of the Context's would use the stream which results in interleaving work that doesn't need to be.

Reply:
> Are there any drawbacks to constructing additional streams, besides construction time? (I'm assuming the latter is why we don't e.g. construct a new stream at the beginning of every context member function implementation)

No, streams are pretty cheap to construct and destroy. Construct a single stream per context to ensure that calling functions like `get_x_t` or `set_x_t` are synced up with each other and we aren't performing operations independently. Not an issue with how the context is written, but seems best to be safe.